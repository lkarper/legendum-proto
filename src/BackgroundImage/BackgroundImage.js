import React from 'react';
import PropTypes from 'prop-types';

const BackgroundImage = (props) => {
    const { 
        classPrefix, 
        imgUrl, 
        imageAltText, 
        children, 
    } = props;

    return (
        <div className={`${classPrefix}__background-container`}>
            <span 
                className={`${classPrefix}__background-image`}
                role='img'
                aria-label={imageAltText}
                style={{
                    backgroundImage: `url( ${imgUrl} )`,
                    backgroundSize: 'cover',
                    backgroundPositionY: 'bottom',
                    backgroundPositionX: 'center'
                }}
            />
            {children}
        </div>
    );
}

BackgroundImage.defaultProps = {
    classPrefix: '', 
    imgUrl: '', 
    imageAltText: 'Default background.', 
};

BackgroundImage.propTypes = {
    classPrefix: PropTypes.string, 
    imgUrl: PropTypes.string,
    imageAltText: PropTypes.string, 
    children: PropTypes.element, 
};

export default BackgroundImage;
