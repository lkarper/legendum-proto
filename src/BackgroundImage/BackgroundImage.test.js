import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import BackgroundImage from './BackgroundImage';

describe('BackgroundImage component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BackgroundImage />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<BackgroundImage />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
