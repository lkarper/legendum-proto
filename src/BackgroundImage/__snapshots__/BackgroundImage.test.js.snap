// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImage component renders the UI as expected 1`] = `
<BackgroundImage
  classPrefix=""
  imageAltText="Default background."
  imgUrl=""
>
  <div
    className="__background-container"
  >
    <span
      aria-label="Default background."
      className="__background-image"
      role="img"
      style={
        Object {
          "backgroundImage": "url(  )",
          "backgroundPositionX": "center",
          "backgroundPositionY": "bottom",
          "backgroundSize": "cover",
        }
      }
    />
  </div>
</BackgroundImage>
`;
